fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkimport/ChunkReader.java

```diff
@@ -167,12 +152,46 @@ public Object readItem() throws Exception {
 
     @Override
     public void open(Serializable checkpoint) throws Exception {
+        if (BulkImportDataSourceStorageType.from(dataSourceStorageType).equals(BulkImportDataSourceStorageType.AWSS3)
+                || BulkImportDataSourceStorageType.from(dataSourceStorageType).equals(BulkImportDataSourceStorageType.IBMCOS)) {
+            cosClient = BulkDataUtils.getCosClient(cosCredentialIbm, cosApiKeyProperty, cosSrvinstId, cosEndpintUrl, cosLocation);
+            if (cosClient == null) {
+                logger.warning("open: Failed to get CosClient!");
+                throw new Exception("Failed to get CosClient!!");
+            } else {
+                logger.finer("open: Got CosClient successfully!");
+            }
+        }
+
         if (checkpoint != null) {
             ImportCheckPointData checkPointData = (ImportCheckPointData) checkpoint;
             importPartitionWorkitem = checkPointData.getImportPartitionWorkitem();
             numOfLinesToSkip = checkPointData.getNumOfProcessedResources();
+            checkPointData.setInFlyRateBeginMilliSeconds(System.currentTimeMillis());
             stepCtx.setTransientUserData(ImportTransientUserData.fromImportCheckPointData(checkPointData));
+        } else {
+            ImportTransientUserData chunkData = new ImportTransientUserData(importPartitionWorkitem, numOfLinesToSkip, importPartitionResourceType);
+            long importFileSize = 0;
+            switch (BulkImportDataSourceStorageType.from(dataSourceStorageType)) {
+            case HTTPS:
+                importFileSize = BulkDataUtils.getHttpsFileSize(importPartitionWorkitem);
+                break;
+            case FILE:
+                importFileSize = BulkDataUtils.getLocallFileSize(importPartitionWorkitem);
+                break;
+            case AWSS3:
+            case IBMCOS:
+                importFileSize = BulkDataUtils.getCosFileSize(cosClient, cosBucketName, importPartitionWorkitem);
+                break;
+            default:
+                throw new Exception ("Doesn't support data source storage type!");
```

Comment:
```suggestion
                throw new Exception("Doesn't support data source storage type!");
```

Reply:
I generally
1. use IllegalStateException in default switch statements that come from enums; and
2. print the name or value of the enum value that caused it in the message

However, neither one very important given that we should never get here unless someone has added a new value to the enum without updating the places its used from.

Reply:
done